### DISCUSSION TITLE
Airflow 2.0.0 db init crashes on windows 10

### DISCUSSION THREAD

**Discussion by gusriobr** [apache/airflow#13448] [General] [answered]
I've managed to install airflow 2.0.0 on windows 10 in a virtual environment with python 3.7 
64b interpreter. 
But when I try to run 
`airflow db init`

I get this error:
 ```
 File "c:\desarrollo\workspaces\wks-py\venvs\airflow\lib\site-packages\airflow\cli\commands\db_command.py", line 24, in <module>
    from airflow.utils import cli as cli_utils, db
  File "c:\desarrollo\workspaces\wks-py\venvs\airflow\lib\site-packages\airflow\utils\db.py", line 26, in <module>
    from airflow.jobs.base_job import BaseJob  # noqa: F401 # pylint: disable=unused-import
  File "c:\desarrollo\workspaces\wks-py\venvs\airflow\lib\site-packages\airflow\jobs\__init__.py", line 22, in <module>
    import airflow.jobs.scheduler_job  # noqa
  File "c:\desarrollo\workspaces\wks-py\venvs\airflow\lib\site-packages\airflow\jobs\scheduler_job.py", line 61, in <module>
    from airflow.utils.dag_processing import AbstractDagFileProcessorProcess, DagFileProcessorAgent
  File "c:\desarrollo\workspaces\wks-py\venvs\airflow\lib\site-packages\airflow\utils\dag_processing.py", line 51, in <module>
    from airflow.utils.process_utils import kill_child_processes_by_pids, reap_process_group
  File "c:\desarrollo\workspaces\wks-py\venvs\airflow\lib\site-packages\airflow\utils\process_utils.py", line 23, in <module>
    import pty
  File "C:\Desarrollo\tools\python-3.7.9-64b\Lib\pty.py", line 11, in <module>
    import tty
  File "C:\Desarrollo\tools\python-3.7.9-64b\Lib\tty.py", line 5, in <module>
    from termios import *
ModuleNotFoundError: No module named 'termios'
```
In the [tty reference page](https://docs.python.org/3.7/library/tty.html) it clearly says:
_Because it requires the termios module, it will work only on Unix._

So, is airflow 2.0.0 expected to work on windows environments?

alembic==1.4.3
apache-airflow==2.0.0
apache-airflow-providers-ftp==1.0.0
apache-airflow-providers-http==1.0.0
apache-airflow-providers-imap==1.0.0
apache-airflow-providers-sqlite==1.0.0
apispec==3.3.2
argcomplete==1.12.2
asn1crypto==1.4.0
attrs==20.3.0
Babel==2.9.0
cached-property==1.5.2
cattrs==1.1.2
certifi==2020.12.5
cffi==1.14.4
chardet==3.0.4
click==7.1.2
clickclick==20.10.2
colorama==0.4.4
colorlog==4.0.2
commonmark==0.9.1
connexion==2.7.0
croniter==0.3.37
cryptography==2.5
defusedxml==0.6.0
dill==0.3.3
dnspython==2.0.0
docutils==0.16
email-validator==1.1.2
Flask==1.1.2
Flask-AppBuilder==3.1.1
Flask-Babel==1.0.0
Flask-Caching==1.9.0
Flask-JWT-Extended==3.25.0
Flask-Login==0.4.1
Flask-OpenID==1.2.5
Flask-SQLAlchemy==2.4.4
flask-swagger==0.2.13
Flask-WTF==0.14.3
funcsigs==1.0.2
graphviz==0.16
gunicorn==19.10.0
idna==2.10
importlib-metadata==1.7.0
importlib-resources==1.5.0
inflection==0.5.1
iso8601==0.1.13
itsdangerous==1.1.0
Jinja2==2.11.2
json-merge-patch==0.2
jsonschema==3.2.0
lazy-object-proxy==1.4.3
lockfile==0.12.2
Mako==1.1.3
Markdown==3.3.3
MarkupSafe==1.1.1
marshmallow==3.10.0
marshmallow-enum==1.5.1
marshmallow-oneofschema==2.1.0
marshmallow-sqlalchemy==0.23.1
natsort==7.1.0
numpy==1.19.4
openapi-spec-validator==0.2.9
pandas==1.2.0
pendulum==2.1.2
prison==0.1.3
psutil==5.8.0
psycopg2==2.8.6
pycparser==2.20
Pygments==2.7.3
PyJWT==1.7.1
pyrsistent==0.17.3
python-daemon==2.2.4
python-dateutil==2.8.1
python-editor==1.0.4
python-nvd3==0.15.0
python-slugify==4.0.1
python3-openid==3.2.0
pytz==2020.5
pytzdata==2020.1
PyYAML==5.3.1
requests==2.23.0
rich==9.2.0
setproctitle==1.1.10
six==1.15.0
SQLAlchemy==1.3.22
SQLAlchemy-JSONField==1.0.0
SQLAlchemy-Utils==0.36.8
swagger-ui-bundle==0.0.8
tabulate==0.8.7
tenacity==6.2.0
termcolor==1.1.0
text-unidecode==1.3
thrift==0.13.0
typing-extensions==3.7.4.3
tzlocal==1.5.1
unicodecsv==0.14.1
urllib3==1.25.11
Werkzeug==1.0.1
WTForms==2.3.3
zipp==3.4.0

**Comment by XD-DENG** [comment, marked as answer]
Please note Airflow doesn't support Windows.

**Comment by ashb** [comment]
You should be able to run Airflow from within WSL though.

**Reply by drprolegend** [reply]
@ashb i have this error:

ModuleNotFoundError: No module named 'termios'

in this file 
Programs\Python\Python36\lib\tty.py"
